x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -4023,6 +4024,25 @@ public void testRejectRemoteEnrichAfterCoordinatorEnrich() {
         );
     }
 
+    public void testNoStackOverflow() {
+        String query = randomBoolean() ? "row a = 1" : "row a = 1 | eval b = a";
+        for (int i = 0; i < 1900; i++) {
```

Comment:
I guess this will be test VM-size dependent, so it might require some fiddling if that'll ever be changed. It might be worth leaving a note... though maybe also not, as we have a few tests in similar situation.

Reply:
Good point, I'll add a comment, it won't hurt